frontend/src/contexts/AuthContext.js

```diff
@@ -17,31 +19,87 @@ export const useAuth = () => {
 
 export const AuthProvider = ({ children }) => {
     const [currentUser, setCurrentUser] = useState()
+    const [currentUserGroup, setCurrentUserGroup] = useState()
     const [loading, setLoading] = useState(true)
     const auth = getAuth(app)
-
+    const whitelist = []
     const provider = new GoogleAuthProvider()
     provider.setCustomParameters({ prompt: 'select_account' })
+    const toast = useToast()
+
+    const showToast = () =>
+        toast({
+            title: 'Invalid email address',
+            description:
+                'Please use a WATonomous or whitelisted email address.',
+            status: 'error',
+            duration: 4000,
+            isClosable: true,
+        })
 
     const login = () => {
         return signInWithPopup(auth, provider)
+            .then((result) => {
+                const { email } = result.user
+                if (whitelist.includes(email)) return
+                if (!email.endsWith('@watonomous.ca')) {
+                    signOut(auth)
+                    showToast()
+                    throw new Error('Invalid email domain')
+                }
+            })
+            .catch((error) => {
+                console.error(error)
+            })
     }
 
     const logout = () => {
         return signOut(auth)
     }
 
+    const setUserGroup = async (user) => {
+        if (user) {
+            const userEmail = user.email
+            const userWatiam = userEmail.substring(0, userEmail.indexOf('@'))
+            const endpoint = `${process.env.REACT_APP_BACKEND_URL}/group/`
+
+            let searchWithEmail = false
```

Comment:
const searchWithEmail = whitelist.includes(userEmail)
^ this just makes it immediately obvious as opposed to writing it on another line